Ambiguous reference error

@MeisterGitarrenheld :
```haskell
test\Compiler\ParserSpec.hs:19:155: error:
    Ambiguous occurrence `getIdentifier'
    It could refer to either the field `getIdentifier',
                             imported from `Compiler.Ast' at test\Compiler\ParserSpec.hs:3:1-29
                          or the field `getIdentifier',
                             imported from `Compiler.Ast' at test\Compiler\ParserSpec.hs:3:1-29
                          or the field `getIdentifier',
                             imported from `Compiler.Ast' at test\Compiler\ParserSpec.hs:3:1-29
```

--- a/package.yaml
+++ b/package.yaml
@@ -7,7 +7,9 @@ copyright: No one
 author: Morty Smith <morty@smith.net>
 maintainer: Rick Sanchez <rick@sanchez.net>
 
-ghc-options: -Wall
+ghc-options: -Wall -fno-warn-name-shadowing
+
+default-extensions: DuplicateRecordFields
 
 dependencies:
  - base == 4.*

--- a/src/Compiler/Ast.hs
+++ b/src/Compiler/Ast.hs
@@ -3,8 +3,6 @@
 --
 -- This module contains the AST
 
-{-# LANGUAGE DuplicateRecordFields #-}
-
 module Compiler.Ast where
 
 -- | 'Identifier' is a simple identifier, e.g. for a parameter

--- a/test/Compiler/ParserSpec.hs
+++ b/test/Compiler/ParserSpec.hs
@@ -11,9 +11,21 @@ main = hspec spec
 
 spec :: Spec
 spec = do
-  describe "program" $ do
-     it "parses an empty class" $ do
-       (parseTestString "class Test {}") `shouldBe` Just [(Class (Identifier "Test") [] [])]
-  describe "objpgrm" $ do
-     it "parses an object class" $ do
-       (parseTestString "class Objectclass {void objectMethod(){new Objectclass();}}") `shouldBe` Just [Class (Identifier "Smallobjectclass") [] [Method {getIdentifier = Identifier "objectMethod",getMods = [],getReturnType = JVoid,getParamList = [], getBody = Just (Block [StmtExprStmt (Instantiation (Name {path = [], getIdentifier = Identifier "Smallobjectclass"}) [])])}]]
+  describe "Parser" $ do
+     it "parses an empty class" $ parseTestString "class Test {}"
+       `shouldBe`
+       Just [Class (Identifier "Test") [] []]
+     it "parses an object class" $ parseTestString "class Objectclass {void objectMethod(){new Objectclass();}}"
+       `shouldBe`
+       Just [Class (Identifier "Objectclass") []
+               [Method { getIdentifier = Identifier "objectMethod"
+                       , getMods = []
+                       , getReturnType = JVoid
+                       , getParamList = []
+                       , getBody = Just (
+                           Block [StmtExprStmt
+                                  (Instantiation
+                                   Name { path = []
+                                        , getIdentifier = Identifier "Objectclass"}
+                                    []
+                                  )])}]]

--- a/tiny-java.cabal
+++ b/tiny-java.cabal
@@ -2,7 +2,7 @@
 --
 -- see: https://github.com/sol/hpack
 --
--- hash: 2b7bef5d98951765f9f46f21335026df995b668f8bfe9b4095382e103b4dc54e
+-- hash: 44031a34a911e773bf54e6519f3236a4260659b4d8ae60fbecdddd395f58df90
 
 name:           tiny-java
 version:        0.0.0
@@ -18,7 +18,8 @@ cabal-version:  >= 1.10
 library
   hs-source-dirs:
       src
-  ghc-options: -Wall
+  default-extensions: DuplicateRecordFields
+  ghc-options: -Wall -fno-warn-name-shadowing
   build-depends:
       base ==4.*
     , megaparsec ==6.*
@@ -36,7 +37,8 @@ test-suite spec
   main-is: Spec.hs
   hs-source-dirs:
       test
-  ghc-options: -Wall
+  default-extensions: DuplicateRecordFields
+  ghc-options: -Wall -fno-warn-name-shadowing
   build-depends:
       base ==4.*
     , hspec ==2.*

